config.go

```diff
@@ -0,0 +1,89 @@
+package quic
+
+import "github.com/lucas-clemente/quic-go/internal/protocol"
+
+// Clone clones a Config
+func (c *Config) Clone() *Config {
```

Comment:
If you're copying every value, why not just `copy := *c; return &copy`?

Reply:
Yes, that's a lot nicer. I was just copying what the standard library does for the `tls.Config`.